xwiki-rendering-syntaxes/xwiki-rendering-syntax-annotatedxhtml/src/main/java/org/xwiki/rendering/internal/renderer/xhtml/AnnotatedXHTMLChainingRenderer.java

```diff
@@ -34,6 +36,8 @@
  */
 public class AnnotatedXHTMLChainingRenderer extends XHTMLChainingRenderer
 {
+    private static final String METADATA_BLOCK_ELEMENT = "span";
```

Comment:
The element used to hold the meta data must depend on the wrapped / inner / nested content. ``span`` is good only if the wrapped content is in-line, otherwise, for block-level content, you must use ``div``. Imagine that the macro outputs a ``table``, it wouldn't be valid to have the ``table`` inside a ``span``.

Reply:
OK it's actually what I thought: then I have to manage a context in the AnnotatedXHTMLChainingRenderer to flag if I'm between begin/endMacroMarker and what is the inline value, right?